src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -126,25 +130,61 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
     param5?: number,
     param6?: number,
     param7?: number
-  ): void {
-    const command: Message.CommandLong = {
-      type: MAVLinkType.COMMAND_LONG,
-      param1: param1 ?? 0,
-      param2: param2 ?? 0,
-      param3: param3 ?? 0,
-      param4: param4 ?? 0,
-      param5: param5 ?? 0,
-      param6: param6 ?? 0,
-      param7: param7 ?? 0,
-      command: {
-        type: mav_command,
-      },
-      target_system: this.currentSystemId,
-      target_component: 1,
-      confirmation: 0,
-    }
+  ): Promise<CommandAck> {
+    return new Promise((resolve, reject) => {
+      // Construct the command
+      const command: Message.CommandLong = {
+        type: MAVLinkType.COMMAND_LONG,
+        param1: param1 ?? 0,
+        param2: param2 ?? 0,
+        param3: param3 ?? 0,
+        param4: param4 ?? 0,
+        param5: param5 ?? 0,
+        param6: param6 ?? 0,
+        param7: param7 ?? 0,
+        command: {
+          type: mav_command,
+        },
+        target_system: this.currentSystemId,
+        target_component: 1,
+        confirmation: 0,
+      }
+
+      this.write(command)
+      let ackReceived = false
+
+      const ackHandler = (commandAck: CommandAck): void => {
+        console.log('Received command acknowledgment:', commandAck)
+        if (commandAck.command.type === mav_command) {
+          ackReceived = true
+          this.onCommandAck.remove(ackHandler)
 
-    this.write(command)
+          if (commandAck.result.type !== MavResult.MAV_RESULT_ACCEPTED) {
+            this._statusText.text = `Command: ${mav_command} failed with result: ${commandAck.result.type}`
+            this._statusText.severity = AlertLevel.Error
+            this.onStatusText.emit()
+            reject(new Error(`Command failed with result: ${commandAck.result.type}`))
```

Comment:
We should receive an argument on the `sendCommandAck` to allow such validation done automatically, or allow the code to deal with the result manually.
It would be nicer if the returned object is not `commandAck` but a wrapper that allows dealing with it in a better way, including raising error/warning messages.

Reply:
> We should receive an argument on the `sendCommandAck` to allow such validation done automatically, or allow the code to deal with the result manually.

Are you thinking of having another method that calls `sendCommandLong` and does the ack listening after the `sendCommandLong` method returns? 

> It would be nicer if the returned object is not `commandAck` but a wrapper that allows dealing with it in a better way, including raising error/warning messages.

What methods do you think should go in the object? Are there any other objects that are similar that I can use for reference?

Reply:
> Are you thinking of having another method that calls `sendCommandLong` and does the ack listening after the `sendCommandLong` method returns?

I'm talking about having just an optional argument for the `sendCommandLong`, for now that should be enough.



> What methods do you think should go in the object? Are there any other objects that are similar that I can use for reference?

A function on the `commandAck` should be enough, an `alert()` that emits an `Swal.fire` would be nice.

Reply:
I didn't know about the `Swal.fire` that seems better than as a text alert! I will add that and remove the current text alert I will also add the optional argument as well

Reply:
Let me know what you think of the changes. I can tweak as needed

Reply:
Just putting my 2 cents (I didn't check the whole logic as patrick is already taking care of that):

I think it's important that the one invoking the command method is informed about the result of the promise (even for timeouts), and should also have the option to decide whether or not a Swal popup should be fired in case of failure or timeout. Different interfaces will have different approaches on that. Some will prefer the automatic handling, and others will need more control.

Reply:
I added another optional argument to suppress swals based on this